### PR TITLE
sol-277 COVID-19 Samoa: Adding sub_district level between district and village level

### PR DESCRIPTION
**Pull request by nick-beyondessential**
### Changes:
- import sub_divisions into Samoa for Covid-Samoa project (Only exist in this project) This was done via migration rather than admin panel because:
  a. One of the districts (Unknown District has no geolocation, which hides from map but breaks admin import)
  b. avoid timing hassles with mixed import/migration of entity_relations
- migrate Samoa villages in Covid-Samoa to District(sub-district) parents.
- Remove facilities from Covid-Samoa
- Add household entity type
- import household entities and add hierarchy relations
- remove facility overlays

### CODE REVIEW

File: packages/database/src/migrations/20210803035205-CovidSamoaAddVillageEntityRelations-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  // console.log('villageRelations: ', villageRelations);
+  const villageCodes = Object.keys(villageRelations);
+  const subDistrictCodes = Object.values(villageRelations);
+
+  const samoaVillageRows = (
+    await db.runSql(`
+      select code, id from entity where "code" in (${arrayToDbString(villageCodes)});
+    `)
+  ).rows;
+  const samoaVillageIds = Object.assign({}, ...samoaVillageRows.map(x => ({ [x.code]: x.id })));
```
Comment (rohan-bes) [inline packages/database/src/migrations/20210803035205-CovidSamoaAddVillageEntityRelations-modifies-data.js:30]: Oh this is a nifty trick!!! I usually use a reduce in this scenario but yours seems quicker!
Gotta be careful tho with spread operator in function arguments, can lead to maximum call stack limit exceeded when dealing with large arrays (10K +)

**Review by rohan-bes — APPROVED** [review]
Nice work, LGTM

**Comment by rohan-bes** [comment]
@nick-beyondessential have we tested these migrations on a feature instance to see if there's any issues? (it seems safe tho)

File: packages/database/src/migrations/20210803035205-CovidSamoaAddVillageEntityRelations-modifies-data.js (hunk 1 above)
Comment (nick-beyondessential) [reply packages/database/src/migrations/20210803035205-CovidSamoaAddVillageEntityRelations-modifies-data.js:30]: Ah good to know!

**Comment by nick-beyondessential** [comment]
Will push through to testing. Which I assume when passed we'll need a new hotfix PR

**Comment by nick-beyondessential** [comment]
@rohan-bes added household entity type.

**Comment by nick-beyondessential** [comment]
Closed in favour of https://github.com/beyondessential/tupaia/pull/2957

**Pull request closed without merging**